### ISSUE TITLE
Move grade question below contributor questions

### ISSUE THREAD

**Issue by karyon** [fsr-itse/EvaP#605]
Currently the grade question is part of the general questions and thus appears above the contributor questions. IMO it should be the very last question to be answered, so evap should make it possible to arrange questions accordingly.

maybe the "Miscellaneous" question should also appear at the end.

**Comment by He3lixxx** [comment]
@karyon, @janno42: Do you see any reasons not to add a "show_below_contributor_questions" boolean to the questionnaire class which can be set to true in order to achieve this? If so, do you have better ideas?

**Comment by janno42** [comment]
i would do it just like that 👍

**Comment by karyon** [comment]
yeah, that would work.

looks like you thought of this already but nevertheless: i think it would be much easier to make this an attribute of the questionnaire instead of single questions.

the order which the questionnaires have right now is hidden in the questionnaire's form and set solely through the drag'n'drop UI in the questionnaire overview. consequently, we might need a new section in that overview, otherwise you would have the questionnaires below and above the contributor questions in the same list. the order in that list would be partly overridden by the new boolean option, which is not nice.

so maybe, with that new section, it might be a cool additional feature to be able to drag and drop the questionnaires between those two sections. sortable.js which we use there does support this, but this might need work to work well with django-dynamic-formset. if we did this, we could completely hide the boolean from the user, just like we do with the ordering parameter.

an idea regarding how to store this: Instead of the boolean, we could introduce ranges to this order, e.g. 0-1000 is for general questionnaires, 1000-2000 for contributors, 2000-3000 for the stuff below that. not sure if it's better in the end, it might simplify stuff in some places but make querysets more complex because you wouldn't check for "show_below_contributor_questions is True" but "order between SOME_CONSTANT and SOME_OTHER_CONSTANT".

long story short, i think the new section in the questionnaire overview is kind of required for the feature to be complete, since the UI is kind of broken without it. but you should probably implement it at the end. the dragging thing is a nice to have for me. the ranges in the ordering i'm not sure about, if i were you i would start with whatever feels better to you and see how it works out.

**Comment by He3lixxx** [comment]
So, after discussing this again, we came to the conclusion that we want to do it like karyon proposed.

Questionnaire: index attribute gets renamed to "order".
- 0-999 means "top" question
- 1000-1999 means contributor question
- 2000-2999 means bottom question

We will add a new section to the questionnaire edit page where all the bottom questionnaires are displayed. If integrating drag and drop with multiple sections is easy, we do it that way. Also, we'll keep the "fake" bottom attribute on the questionnaire edit page. If a user changes the section with the check box, the question should just be appended to the other questions in the section.

The changes made to this point need to be adapted accordingly (top / bottom list comprehension e.g.)

**Comment by He3lixxx** [comment]
Aaaaand we decided to do it differently. Since the magic values are awkward and we'd want to keep the is_for_contributors attributor although we can't really use it for sorting, we decided to do it with two attributes. One is a char field which indicates "top questionnaire", "contributor questionnaire" or "bottom questionnaire". Questionnaires are sorted primarily by this field. Also, each questionnaire keeps the order attribute which decides their order in the same questionnaire category.
